testing/unittests/jlibtests.cpp

```diff
@@ -2113,20 +2113,129 @@ element1: !el scalarvalue
+            Owned<IPropertyTree> json = createPTreeFromJSONString(jsonEmptyNameMarkup, ipt_none, ptr_ignoreWhiteSpace, nullptr);
 
-        //if we want the final compares to be less fragile (test will have to be updated if formatting changes) we could reparse and compare trees again
-        StringBuffer ml;
-        toJSON(json, ml.clear(), 2, JSON_Format|JSON_SortTags|JSON_HideRootArrayObject);
-        CPPUNIT_ASSERT(streq(ml, jsonMarkup));
+            //if we want the final compares to be less fragile (test will have to be updated if formatting changes) we could reparse and compare trees again
+            StringBuffer ml;
+            toJSON(json, ml.clear(), 2, JSON_Format|JSON_SortTags|JSON_HideRootArrayObject);
+            CPPUNIT_ASSERT(streq(ml, jsonEmptyNameMarkup));
+
+            testPtreeEncode("@a/b*c=d\\e.f", "@a_fb_x2A_c_x3D_d_be.f");
+            testPtreeEncode("A/B*C=D\\E.F", "A_fB_x2A_C_x3D_D_bE.F");
+            testPtreeEncode("_x123_blue.berry", "__x123__blue.berry");
+            testPtreeEncode("_>>_here_**", "___x3E3E___here___x2A2A_");
+            testPtreeEncode("出/售\\耐\"'久", "出_f售_b耐_Q_q久");
+            testPtreeEncode("@出售'\"耐久", "@_xE587BAE594AE__q_Q_xE88090E4B985_");
+            testPtreeEncode("出售耐久", "出售耐久");
+            testPtreeEncode("@出/售\\耐'久", "@_xE587BA__f_xE594AE__b_xE88090__q_xE4B985_");
+            testPtreeEncode("出/售\\耐'久", "出_f售_b耐_q久");
+            testPtreeEncode("space space", "space_sspace");
+
+            static constexpr const char * jsonMarkup = R"!!(
+{
+    "Nothing_ToEnodeHere": {
+        "@attributes_need_unicode_encoded_出售耐久": "encode",
+        "but_elements_support_it_出售耐久": "no encode",
+        "出\\n售耐/久@Aa_a": {
+        "@\\naa_bb食品并/林": "atvalue",
+        "abc/d @ \"e'f": ["value"],
+        "": "noname"
+        }
+    }
+}
+)!!";
+
+            static constexpr const char * yamlMarkup = R"!!(
+Nothing_ToEnodeHere:
+  attributes_need_unicode_encoded_出售耐久: "encode"
+  but_elements_support_it_出售耐久: ["no encode"]
+  "出\\n售耐/久@Aa_a":
+    "\\naa_bb食品并/林": "atvalue"
+    "abc/d @ \"e'f": ["value"]
+    "": [noname]
+)!!";
+
+            static constexpr const char * toXMLOutput = R"!!(<Nothing_ToEnodeHere attributes__need__unicode__encoded___xE587BAE594AEE88090E4B985_="encode">
+ <but_elements_support_it_出售耐久>no encode</but_elements_support_it_出售耐久>
+ <出_bn售耐_f久_aAa__a _bnaa__bb_xE9A39FE59381E5B9B6__f_xE69E97_="atvalue">
+  <_0>noname</_0>
+  <abc_fd_s_a_s_Qe_qf>value</abc_fd_s_a_s_Qe_qf>
+ </出_bn售耐_f久_aAa__a>
+</Nothing_ToEnodeHere>
+)!!";
+
+            static constexpr const char * toJSONOutput = R"!!({
+ "Nothing_ToEnodeHere": {
+  "@attributes_need_unicode_encoded_出售耐久": "encode",
+  "but_elements_support_it_出售耐久": "no encode",
+  "出\n售耐/久@Aa_a": {
+   "@\naa_bb食品并/林": "atvalue",
+   "": "noname",
+   "abc/d @ "e'f": "value"
+  }
+ }
+})!!";
+
+            static constexpr const char * toYAMLOutput = R"!!(Nothing_ToEnodeHere:
```

Comment:
Was this supposed to say "NothingToEncodeHere" ?

(and in various other places)?

Reply:
It was the "Enode" rather than "Encode" that I was commenting on